env-config.js

```diff
@@ -118,5 +119,12 @@ export class Config {
     this.key = configData.key || '';
     this.account = configData.account || 'http://services.lblod.info/diff-consumer/account';
     this.account_graph = configData.account_graph || 'http://mu.semte.ch/graphs/diff-producer/login';
+
+    // Apply configuration overrides, if any exist
+    if (configOverrideData != undefined) {
+      for (let key in configOverrideData) {
+        this[key] = configOverrideData[key];
```

Comment:
This will only merge configs that are one level deep.
But what will happen in the future if we extend the config with nested levels?
Suppose e.g. the original config
```
{
  //the well known previous parameters
  "theNewCoolConfigParam": {
    "theParameter": "bar"
  }
}
```

Suppose then we have the following `override` config:
```
{
  "theNewCoolConfigParam": {
    "anotherNewExtraParam": "baz"
  }
}
```
What do we want to happen?
Ignore it the original config? Merge both?
In case of arrays; merging becomes even less clear. (e.g. what if you meant; yeah the first element needs to be replaced; and add the second one)

I think it can get complicated; maybe not from the technical side (there are libraries for that);
    rather make the behaviour clear to someone setting the maintainer up.

So I would go for the simple approach, ignore the 'original' config. So basically your suggested code won't change :-)
BUT BUT, it would make a very visible `console.warn`, by checking

```
if(typeof(configOverrideData[key]) == "object") {
 //warn the user and print the config that's going to be used
}
```

Reply:
BTW: is there no place we print the full config used for the specific stream?

Reply:
Good point about nested overrides; I'll add it to the readme in addition to warning the user in case the override contains some sort of object.

Reply:
> BTW: is there no place we print the full config used for the specific stream?

The line [here](https://github.com/lblod/delta-producer-publication-graph-maintainer/blob/c1e4902c964e085ec9817caf952535d91b0a9cee/app.js#L30) prints out the configuration for each stream one by one.

Reply:
I refactored the logging a bit to have the configuration output and the possible warning (if there is an object value) in the same place.